### ISSUE TITLE
runtime new value provided to cf.environment does not take effect

### ISSUE THREAD

**Issue by naruraghavan** [pivotalservices/ya-cf-app-gradle-plugin#12]
I have the following snippet of JenkinsFile:

```
        stage('cf-push to Stage') {
            when { expression { return env.JOB_NAME == 'blah blah' } }
            steps {
                //unstash 'appJar'
                withCredentials([
                        [$class: 'UsernamePasswordMultiBinding', credentialsId: "cf_push_prod", usernameVariable: 'CF_USER', passwordVariable: 'CF_PWD']
                ]) {
                    step(
                            script {
                                sh "./gradlew cf-push -Pcf.ccUser=\${CF_USER} -Pcf.ccPassword=\${CF_PWD} -Pcf.filePath=${application}/build/libs/${appJarName}.jar -Pcf.ccHost=blah -Pcf.org=NOW -Pcf.space=STAGE -Pcf.domain=blah -Pcf.environment='[\"JAVA_OPTS\": \"-Djava.security.egd=file:///dev/urandom\", \"SPRING_PROFILES_ACTIVE\": \"stage\"]' -Pcf.name=blah -Pcf.hostName=blah"
                            }
                    )
                }
            }
            post {
                success {
                    println "JOB_NAME=${env.JOB_NAME} Ended Successfully"
                }
            }
        }
```

Whatever is on the `cfConfig`:
```
    environment = ["JAVA_OPTS": "-Djava.security.egd=file:///dev/urandom", "SPRING_PROFILES_ACTIVE": "dev"]
```
is what is being pushed to user defined variable and the override provided thru the gradlew parameter `cf.environment` is not taking effect.

**Comment by bijukunjummen** [comment]
Will look at it today @naruraghavan , sorry for the delay.

**Comment by markphahn** [comment]
I have a similar issue. In my build.gradle file I have:
`    name = project.name + "-" + space"`
but I cannot overide the name via the command line:
`gradle cf-push -Pcf.space=test`
-MpH

**Comment by bijukunjummen** [comment]
Nice, happy to see that the project is being used @markphahn , let me open a separate issue for you though @markphahn - since the space internally is set/overridden
 a little differently.

**Comment by bijukunjummen** [comment]
I have a fix @naruraghavan. Will cut a release later in the day. I see that you have closed the issue - did you find some other workaround?

**Comment by naruraghavan** [comment]
No I did not close the issue. Still waiting for a fix - should I use RC1?

**Comment by bijukunjummen** [comment]
Yes, there is a fix in `2.0.0-RC1` @naruraghavan  if you can give it a try, to override or specify the environment variable you have to do this:
`./gradlew cf-push .... -Pcf.environment.SPRING_PROFILES_ACTIVE=stage` which will override the one you have set through `cfConfig` block. Also just a note, there is a breaking change with this version (hence the major version change), the `hostName` is now called `host` if you are setting this attribute.

**Comment by naruraghavan** [comment]
this works! Perfect!
